### DISCUSSION TITLE
[Idea] Optional ternary alternative

### DISCUSSION THREAD

**Discussion by ThreeSevenths** [dotnet/csharplang#7504] [Language Ideas]
### Proposal

Make the alternative portion of a ternary operator optional. When not provided, will be `default`.

### Syntax

```c#
conditional_expression ? statement
``` 

Examples:
```c#
string? maybeMaybe = assignstring ? "Maybe";
Response.DebugDiagnostics = config?.ReturnDiagnostics == true ? context.DebugDiagnostics;
```

### Execution

At compile-time, the compiler will assume that any missing alternative will implicitly become default. The existing semantics for typing the result of the ternary remain in place. Any existing semantics regarding target typing will consider only the consequent of the operator and the assignability to the return type (eg. CS8600 would still apply). 

I do not believe this would change #33 or #881 since it's only a compile time convenience.

**Comment by HaloFour** [comment]
This creates ambiguities in the language as ternary expressions can be nested.

**Comment by ThreeSevenths** [comment]
Yes, if you provided an alternative in a nested ternary, the compiler would not know how to resolve which operator to apply it to. It could try and be clever and deduce this based on the return type of the alternative, however I'd be happy enough to either restrict this to unambiguous cases or require parentheses around the statement to resolve this. The additional usefulness of the optional alternative in those cases would be minimal, this would primarily benefit the simple ternary assignment syntax exemplified above.

**Reply by ThreeSevenths** [reply]
It might also be acceptable to assume that the most outer ternary would be the one the alternative applies to.

Example:
```c#
var maybeMaybe = condition1 ? condition2 ? "Maybe" : "NotCondition1"; 

//would become

var maybeMaybe = condition1 ? condition2 ? "Maybe" : default : "NotCondition1";

```

**Reply by HaloFour** [reply]
Even if the compiler had specific rules as to how to determine the default case, that doesn't make it any easier for the reader who has to internalize and apply those parsing rules.  Being more explicit, while more verbose in these edge cases, is simply clearer.

**Reply by ThreeSevenths** [reply]
I agree with you that being explicit in an edge case is better. In the simple case, I see a benefit to making conditional assignments cleaner.

**Comment by jnm2** [comment]
I prefer seeing the extra characters `: default` (or `: null` or `: 0`) for this situation.
